server-spi-private/src/main/java/org/keycloak/models/Constants.java

```diff
@@ -73,5 +74,12 @@
 
     int DEFAULT_MAX_RESULTS = 100;
 
+    // Delimiter to be used in the configuration of authenticators (and some other components) in case that we need to save
+    // multiple values into single string
+    String CFG_DELIMITER = "##";
```

Comment:
`public static`

Since `Constants` is an interface, this is not an option, and the Pattern below would be instantiated with every `Constants` instantiations. Could this be placed rather to a proper class or - even better - `Constants` be turned into a final class?

Reply:
I don't think this will be the case as fields declared on interface are automatically considered with modifiers "public static final" . Even if there are some instantiations of objects, which inherit from Constants interface, the pattern will be compiled just once. However AFAIK "constant interface" is not good pattern and I don't like it too, so I treat this as an opportunity to change this to "final class" :)

Reply:
You're indeed right, I should stop doing reviews that late in the evening :-) Double thanks for turning this interface into a final class!